packages/server/src/utilities/rowProcessor/attachments.ts

```diff
@@ -7,22 +7,71 @@ import {
   Table,
 } from "@budibase/types"
 import { ObjectStoreBuckets } from "../../constants"
+import { getRowParams } from "../../db/utils"
 
 export class AttachmentCleanup {
-  static async coreCleanup(fileListFn: () => string[]): Promise<void> {
+  static async coreCleanup(
+    fileListFn: () => string[] | Promise<string[]>,
+    opts: { tableId?: string; schema?: Table["schema"] } = {}
+  ): Promise<void> {
+    let files = await Promise.resolve(fileListFn())
+    if (files.length === 0) {
+      return
+    }
+
     const appId = context.getWorkspaceId()
     if (!dbCore.isProdWorkspaceID(appId)) {
       const prodAppId = dbCore.getProdWorkspaceID(appId!)
-      // if prod exists, then don't allow deleting
       const exists = await dbCore.dbExists(prodAppId)
       if (exists) {
-        return
+        files = await AttachmentCleanup.excludeFilesUsedInProd(
+          files,
+          prodAppId,
+          opts
+        )
+        if (files.length === 0) {
+          return
+        }
       }
     }
-    const files = fileListFn()
-    if (files.length > 0) {
-      await objectStore.deleteFiles(ObjectStoreBuckets.APPS, files)
+
+    await objectStore.deleteFiles(ObjectStoreBuckets.APPS, files)
+  }
+
+  private static async excludeFilesUsedInProd(
+    files: string[],
+    prodAppId: string,
+    opts: { tableId?: string; schema?: Table["schema"] }
+  ): Promise<string[]> {
+    const { tableId, schema } = opts
+    if (!tableId || !schema) {
+      return files
+    }
+
+    const prodDb = dbCore.getDB(prodAppId)
+    const response = await prodDb.allDocs(
```

Comment:
The performance here could be an issue. 

As the main problem lies with the app exports and unwanted lingering attachment files, it may be better to do a cleanup on export?